charts/secrets-store-sync-controller/README.md

```diff
@@ -0,0 +1,36 @@
+# Deploying the Secrets Store Sync Controller
+You can deploy the Secrets Store Sync Controller using Helm. This guide provides instructions for deploying the Secrets Store Sync Controller using Helm.
+
+You can use the following commands to deploy the Secrets Store Sync Controller using Helm:
+```sh
+helm install -f values <path_to_values.yaml> secrets-store-sync-controller charts/secrets-store-sync-controller
+```
+
+# Configuration and Parameters
+You can customize the installation by modifying values in the values.yaml file or by passing parameters to the helm install command using the --set key=value[,key=value] argument.
+
+| Parameter Name                                   | Description                                                                                       | Default Value                                                                                                                                                                         |
+|--------------------------------------------------|---------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `providerContainer`                              | The container for the Secrets Store Sync Provider                                                 | `[- name: provider-aws-installer ...]`                                                                                                                                                |
+| `controllerName`                                 | The name of the Secrets Store Sync Controller.                                                    | `secret-sync-controller-manager`                                                                                                                                                      |
+| `namespace`                                      | The namespace to deploy the chart to.                                                             | `secret-sync-controller-system`                                                                                                                                                       |
```

Comment:
Do we need a helm variable for namespace? We can just use `Release.Namespace` instead? (xref: https://helm.sh/docs/chart_template_guide/builtin_objects/)

Reply:
user has to provide --namespace while installing the helm chart. I don't think this is a good UX as this is error prone. Even if we add documentation, it's likely that users might miss it. Moreover for "default" ns this will still fail with below error:
```shell
Error: INSTALLATION FAILED: Unable to continue with install: Namespace "default" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "secrets-store-sync-controller"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"
```

Reply:
The recommendation should be to use the namespace flag as part of helm installation (xref: https://github.com/helm/helm/issues/5465#issuecomment-473942223). By adding namespace creation in the helm template, we'll force users to only deploy the controller in newly created namespaces which is incorrect. For instance, they can't deploy the controller in `kube-system` namespace now because there is a namespace template here.

Helm already supports creating namespace with `--create-namespace` flag, so that's what we should use.

1. Remove the namespace template
2. Remove namespace from the values and other usages

Here is WI charts for reference: https://github.com/Azure/azure-workload-identity/tree/main/charts/workload-identity-webhook#install-chart